### DISCUSSION TITLE
Can GUI layer of CSLA run on Windows DotNet v5 and BL and DataPortal layer on BSD mono?

### DISCUSSION THREAD

**Discussion by aprokopyev** [MarimerLLC/csla#2047] [Questions]
Hello,

Please let me know, can different layers of CSLA application run on different OSes and different versions and even implementations of runtime like DotNet v5 on Windows, DotNet v5 on Linux, Mono v6.12 on BSD for example on OpenBSD?

If it is not possible then which restrictions apply for interaction between different layers of CSLA on different OSes, IL versions and implementations?

It seems to be already partially answered in the thread:
https://github.com/MarimerLLC/cslaforum/issues/844#issuecomment-561792429
If there is nothing new about this question, please just ignore it.

Thank you very much.

**Comment by rockfordlhotka** [comment]
Yes, you can create your client app for Windows (using WPF, UWP, Windows Forms, or Xamarin) and run your backend data portal services on Linux or any other operating system that can run net5.0.

Nothing special is required to make this happen, it should just work.

**Reply by rockfordlhotka** [reply]
You do specify mono, and I _think_ that is ok, as long as the current version of mono is compatible with netstandard2.0 (and I think it is?).
